resolwe/flow/migrations/0004_data_dependency_2.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.7 on 2017-09-21 10:43
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('flow', '0003_data_dependency_1'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='datadependency',
+            name='kind',
+            field=models.CharField(choices=[('io', 'Input/output dependency'), ('subprocess', 'Subprocess')], default='io', max_length=16),
```

Comment:
We currently set the dependency kind for all existing dependencies to `io`, even though this is not entirely correct (we previously treated both dependencies as IO, but abused them in workflows and spawned processes).

We could add a data migration to correct this, but it would be quite complex in order to handle IO dependencies, workflows and spawned processes.

Reply:
You can check inputs of the child process and if parent is there it is IO dependency and subprocess dependency otherwise.

Reply:
Sure, the question is do we even need to care about old dependencies?

Reply:
I think that we should keep the state consistent, no matter when the object were created. Otherwise it can cause some strange behaviour in the future.

Reply:
> You can check inputs of the child process and if parent is there

This of course also requires duplicating the input processing code in the migration, which is IMO what makes it a bit more complex as we either need to depend on some utility functions in resolwe modules (less desirable as those functions may change later) or copy everything into the migration (so it is a snapshot in time).

Reply:
But I think that it is worth keeping relations properly labeled. You only have to copy `iterate_fields` function and it will get deleted once migrations will be squashed.

Reply:
Ok, I'll copy it over.

Reply:
Data migration added and tested on a QA2 database dump.